cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -1527,4 +1628,26 @@ private void emitTimeMetrics(String namespace, String appName, String metricName
     long timeTaken = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startTime);
     metricsCollectionService.getContext(tags).gauge(metricName, timeTaken);
   }
+
+  public void markAppsAsLatest(List<ApplicationId> appIds) throws Exception {
+    // Validate the appIds do not contain duplicates, i.e. we are not trying to mark
+    // multiple versions of the same application as latest
+    Set<String> validated = new HashSet<>();
+    for (ApplicationId appId: appIds) {
+      String appName = appId.getApplication();
+      if (validated.contains(appName)) {
+        throw new Exception(String.format(
+            "Marking multiple versions of an application (%s) as latest is not supported.",
+            appName
+        ));
+      }
+      validated.add(appName);
```

Comment:
Set.add() returns a boolean. You can add it and check if it already exists in the same method call instead of having a separate contains() check above.

Reply:
updated.